apps/web/lib/features/daily-plan/all-plans-modal.tsx

```diff
@@ -188,13 +200,19 @@ export const AllPlansModal = memo(function AllPlansModal(props: IAllPlansModal)
 										Cancel
 									</Button>
 									<Button
-										disabled={!selectedFuturePlan}
+										disabled={!customDate || createDailyPlanLoading}
 										variant="default"
 										type="submit"
 										className={clsxm('py-3 px-5 rounded-md font-light text-md dark:text-white')}
-										onClick={handleCalendarSelect}
+										onClick={selectedPlan ? handleCalendarSelect : createEmptyPlan}
 									>
-										Select
+										{selectedPlan ? (
+											'Select'
+										) : createDailyPlanLoading ? (
+											<SpinnerLoader variant="light" size={20} />
+										) : (
+											'Add plan'
```

Comment:
@CREDO23 (CC: @Cedric921 ) we need to make sure all "hardcoded" text actually translated in all codebase (i18n)... Please make sure all PRs have that.